### PR TITLE
Port Dimensions fast path + bug fix from upstream

### PR DESCRIPTION
**Pull request by TheGlitch76**
Just some ports from upstream Fabric.

The bugfix mixin isn't strictly necessary, because most people will have FAPI anyway, but I added it in case that two-year-old bug never gets fixed

### CODE REVIEW

File: library/worldgen/dimension/src/main/java/org/quiltmc/qsl/worldgen/dimension/impl/QuiltDimensionsImpl.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 package org.quiltmc.qsl.worldgen.dimension.impl;
 
 import com.google.common.base.Preconditions;
+import net.minecraft.server.network.ServerPlayerEntity;
```
Comment (triphora) [inline library/worldgen/dimension/src/main/java/org/quiltmc/qsl/worldgen/dimension/impl/QuiltDimensionsImpl.java]: This is going to fail checkstyle

Comment (TheGlitch76) [reply library/worldgen/dimension/src/main/java/org/quiltmc/qsl/worldgen/dimension/impl/QuiltDimensionsImpl.java]: oops. will fix in a bit

**Review by peaceheis — COMMENTED** [review]
everything looks good to me!if you can just fix that one style issue, i'll approve.

**Comment by EnnuiL** [comment]
This PR will be necessary in order to quiltify the Fabric Dimensions API on QFAPI, so yeah, we'll actually need it

**Review by LambdAurora — APPROVED** [review]

**Review by EnnuiL — APPROVED** [review]

**Review by peaceheis — APPROVED** [review]

**Pull request merged**